src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,165 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private int size;
+    private Node<T> head;
+    private Node<T> tail;
+
     @Override
     public void add(T value) {
+        if (isEmpty()) {
+            final Node<T> first = head;
+            final Node<T> newNode = new Node<>(null, value, first);
+            head = newNode;
+            tail = newNode;
+        } else {
+            final Node<T> last = tail;
+            final Node<T> newNode = new Node<>(last, value, null);
+            tail = newNode;
+            last.next = newNode;
+        }
+        size++;
```

Comment:
```suggestion
        Node<T> newNode = new Node<>(tail, value, null);
        if (size == 0) {
            head = newNode;
        } else {
            tail.next = newNode;
        }
        tail = newNode;
        size++;
```

we could simplify this